edgelet/hsm-sys/azure-iot-hsm-c/src/certificate_info.c

```diff
@@ -292,347 +232,239 @@ time_t get_utc_time_from_asn_string(const unsigned char *time_value, size_t leng
     return result;
 }
 
-static time_t get_utctime_value(const unsigned char* time_value)
+static int parse_common_name(CERT_DATA_INFO* cert_info, X509* x509_cert)
 {
-    // Check the the type
-    if (*time_value != ASN1_UTCTIME)
+    int result;
+    int cn_size = MAX_LEN_COMMON_NAME + 1;
+    char *cn;
+    /*const*/ X509_NAME* subj_name;
+
+    if ((subj_name = X509_get_subject_name(x509_cert)) == NULL)
     {
-        LogError("Parse time error: Unknown time format");
-        return 0;
+        LOG_ERROR("Failure obtaining certificate subject name");
+        result = __FAILURE__;
     }
-    else
+    else if ((cn = malloc(cn_size)) == NULL)
     {
-        return get_utc_time_from_asn_string((time_value + 2), *(time_value + 1));
+        LOG_ERROR("Could not allocate memory for common name");
+        result = __FAILURE__;
     }
-}
-
-static size_t calculate_size(const unsigned char* buff, size_t* pos_change)
-{
-    size_t result;
-    if ((buff[0] & EXTENDED_LEN_FLAG))
+    else
     {
-        // We are using more than 128 bits, let see how many
-        size_t num_bits = buff[0] & LEN_FLAG_COUNT;
-        result = 0;
-        for (size_t idx = 0; idx < num_bits; idx++)
+        memset(cn, 0, cn_size);
+        if (X509_NAME_get_text_by_NID(subj_name, NID_commonName, cn, cn_size) == -1)
         {
-            unsigned char temp = buff[idx + 1];
-            if (idx == 0)
-            {
-                result = temp;
-            }
-            else
-            {
-                result = (result << 8) + temp;
-            }
+            LOG_INFO("X509_NAME_get_text_by_NID could not parse subject field 'CN'");
```

Comment:
Is this a failure case? You aren't setting `result = __FAILURE__;` here.

Reply:
This failure indicates that the common name does not exist in the certificate. This would be an error by higher level applications consuming this API since the common name would be set to NULL.